core/embed/sec/secret/secret_keys_common.c

```diff
@@ -0,0 +1,126 @@
+/*
+ * This file is part of the Trezor project, https://trezor.io/
+ *
+ * Copyright (c) SatoshiLabs
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#ifdef SECURE_MODE
+
+#include <sec/secret.h>
+#include <sec/secret_keys.h>
+#include <trezor_model.h>
+#include <trezor_rtl.h>
+
+#include "../storage/storage_salt.h"
+#include "hmac.h"
+#include "memzero.h"
+#include "nist256p1.h"
+#include "secret_keys_common.h"
+
+static void diversify_and_derive(uint16_t index, uint16_t subindex,
+                                 uint8_t dest[SHA256_DIGEST_LENGTH],
+                                 uint8_t master_key[SHA256_DIGEST_LENGTH],
+                                 uint8_t master_key_length) {
```

Comment:
Input parameters go first, output parameters go last. Input pointers should be const.
```suggestion
static void diversify_and_derive(uint16_t index, uint16_t subindex,
                                 const uint8_t master_key[SHA256_DIGEST_LENGTH],
                                 uint8_t master_key_length,
                                 uint8_t dest[SHA256_DIGEST_LENGTH]) {
```

Reply:
done in [`bd5c87`](https://github.com/trezor/trezor-firmware/pull/5770/commits/bd5c87f6ce5fc72966890ebe756afb93265e5320)